File: 2015/01/01.py
f = open("input.txt", "r")
seq = f.read()

print("floor is : " , seq.count('(') - seq.count(')'))

floor = 0
idx = 0

for cmd in seq :
    idx = idx + 1

    if cmd == '(' :
        floor = floor + 1
    else :
        floor = floor - 1

    if floor == -1 :
        print("second part answer is : " , idx)
        break

File: 2018/01/01.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>

int main()
{

    std::ifstream infile("input.txt");

    //First part:
    int increment;
    int Freq = 0;
    std::vector<int> incs;

    while (infile >> increment)
    {
        // std::cout << increment << std::endl;
        Freq += increment;
        incs.push_back(increment);
    }
    std::cout << Freq << std::endl;

    //Second part:
    Freq = 0;
    bool Duplicate_found = false;
    std::vector<int> Freqs;
    Freqs.push_back(Freq);

    while (!Duplicate_found)
    {
        for (int increment : incs)
        {
            Freq += increment;
            if (std::count(Freqs.begin(), Freqs.end(), Freq) == 0)
            {
                Freqs.push_back(Freq);
            }
            else
            {
                std::cout << Freq << std::endl;
                Duplicate_found = true;
                break;
            }
        }
    }
    return 0;
}